app.js

```diff
@@ -1,8 +1,154 @@
-// import functions and grab DOM elements
+import { renderGoblin } from './utils.js';
 
-// let state
+const form = document.getElementById('form');
+const goblinList = document.getElementById('goblins');
+const heroHPel = document.getElementById('heroHP');
+const defeatedGoblinsEl = document.getElementById('defeatedGoblins');
 
-// set event listeners 
-  // get user input
-  // use user input to update state 
-  // update DOM to reflect the new state
+
+function playCheer() {
+    new Audio(`assets/18364__jasinski__bb-claprhm.mp3`).play();
+}
+function playHit() {
+    new Audio(`assets/214989__amishrob__baseballhitandcrowdcheer.mp3`).play();
+}
+function playStrike() {
+    new Audio(`assets/45110__sanus-excipio__thats-it.wav`).play();
+}
+
+
+let defeatedGoblins = [];
+let currentID = 6;
+let goblins = [{
+    name: `Randy Johnson`, id: 1, hp: 8
+}, {
+    name: `Terry Pendleton`, id: 2, hp: 7
+}, {
+    name: `Greg Maddux`, id: 3, hp: 5
+}, {
+    name: `Roger Clemens`, id: 4, hp: 6
+}, {
+    name: `Mario Soto`, id: 5, hp: 5
+}];
+let heroHP = 3;
+
+
+form.addEventListener('submit', (e) => {
+    console.log('click');
+    e.preventDefault();
+    const data = new FormData(form);
+
+    const goblinName = data.get('goblin-name');
+
+    const newGoblin = {
+        id: currentID,
+        name: goblinName,
+        hp: Math.ceil(Math.random() * 5),
+    };
+    currentID++;
+
+    goblins.push(newGoblin);
+
+    displayGoblins();
+});
+
+
+alert(`It was but a week ago you found the strange baseball bat thrumming with energy in a dumpster outside an abandoned antique sporting goods store. It called to you like the board game in jumanji. Since then, you've only hit dingers against your friends. only yard bombs. The bat seems... more than just lucky, and you wonder if you could ever strikeout. Thats when you pass by a baseball convention at the local casino and decide to REALLY test it out..`);
+
+function displayGoblins() {
+    goblinList.textContent = '';
+    
+    for (let goblin of goblins) {
+        const goblinEl = renderGoblin(goblin);
+
+        if (goblin.hp <= 0) {
+            goblinEl.style.opacity = '.5';
+        //     goblinEl.classList.add(`animate__animated`, `animate__flip`);
+        }
+        
+        goblinEl.addEventListener('click', () => {
+            console.log('goblin clicked');
+            goblinClick(goblin);
+        });
+        
+        goblinList.append(goblinEl);
+    }
+}
+
+function goblinClick(goblin) {
+
+    if (goblin.hp <= 0) { alert('stop, stop!! ' + goblin.name + ' is already dead!'); 
+        return;
+    }
+
+    const goblinDie = Math.ceil(Math.random() * 6);
+    const heroDie = Math.ceil(Math.random() * 12);
+
+    if (goblinDie > heroDie) {
```

Comment:
This way of writing this function makes it impossible for both the goblin and the player to get hit on the same turn (which is fine, but just an fyi for future game logic). It also means that if the numbers are the same, nothing would happen since there is no heroDie === goblinDie